### DISCUSSION TITLE
Option to delete content when removing a field from a field layout

### DISCUSSION THREAD

**Discussion by carlcs** [craftcms/cms#8441] [Ideas] [answered]
Currently when removing a field from a field layout, the content remains in the db. It would be nice if you were asked whether you want to keep or delete the data.

**Comment by dennisfrank** [comment]
I had this issue today (Craft 2):

After removing a Matrix field from an entry type the previously saved data from this field was still connected to the entry. The content is still there for entry types that no longer have the field.

The Matrix field is connected to a bunch of other entry types and I checked if the field is defined and not empty with a very global condition. I had to write a very specific condition to manage this exception because I couldn’t get the data deleted from the DB.

**Comment by brandonkelly** [comment]
This behavior will change when we implement #1524

**Comment by brandonkelly** [comment, marked as answer]
For the time being, you can use the [`resave/entries`](https://craftcms.com/docs/4.x/console-commands.html#resave-entries) command to clear out existing field values after a field is removed from a field layout:

```sh
php craft resave/entries --section=mySection --type=myType --set=myFieldHandle --to=:empty:
```

**Reply by michtio** [reply]
@brandonkelly this feature is not working for Money fields.
See https://github.com/craftcms/cms/issues/12539
